dts/arm/renesas/smartbond/da1469x.dtsi

```diff
@@ -96,6 +97,57 @@
 			clock-src = <&rc32k>;
 			status = "okay";
 		};
+
+		regulators {
+			compatible = "renesas,smartbond-regulator";
+			vdd: VDD {
+				regulator-init-microvolt = <900000>;
+				regulator-boot-on;
+				regulator-initial-mode = <
+					(DA1469X_MODE_SLEEP_LDO |
+					 DA1469X_MODE_DCDC_HIGH_BATTERY_VOLTAGE_ON |
+					 DA1469X_MODE_DCDC_LOW_BATTERY_VOLTAGE_ON)>;
+			};
+			vdd_clamp: VDD_CLAMP {
+				regulator-boot-on;
+				regulator-always-on;
+				regulator-init-microvolt = <706000>;
+			};
+			vdd_sleep: VDD_SLEEP {
+				regulator-boot-on;
+				regulator-init-microvolt = <750000>;
+			};
+			v14: V14 {
+				regulator-init-microvolt = <1400000>;
+				regulator-boot-on;
+				regulator-initial-mode = <
+					(DA1469X_MODE_DCDC_HIGH_BATTERY_VOLTAGE_ON |
+					 DA1469X_MODE_DCDC_LOW_BATTERY_VOLTAGE_ON)>;
+			};
+			v18: V18 {
+				regulator-init-microvolt = <1800000>;
+				regulator-boot-on;
+				regulator-initial-mode =
+					<DA1469X_MODE_DCDC_HIGH_BATTERY_VOLTAGE_ON>;
+			};
+			v18p: V18P {
+				regulator-init-microvolt = <1800000>;
+				regulator-boot-on;
+				regulator-initial-mode = <
+					(DA1469X_MODE_SLEEP_LDO |
+					 DA1469X_MODE_DCDC_HIGH_BATTERY_VOLTAGE_ON)>;
+			};
+			v30: V30 {
+				regulator-init-microvolt = <3000000>;
+				regulator-boot-on;
+				regulator-initial-mode = <
+					(DA1469X_MODE_SLEEP_LDO |
+					 DA1469X_MODE_LDO_VBAT_ON |
+					 DA1469X_MODE_LDO_VBUS_ON |
+					 DA1469X_MODE_V30_VBAT_CLAMP_ON)>;
```

Comment:
what is this trying to configure? modes are not something to configure any parameters of a regulator, but "operating modes" (e.g. regulation schemes)

Reply:
This sets that V30 rail can be powered from VBUS line (when available) or VBAT line, VBAT line without LDO for low battery condition, and that if should also switch to special clamp mode during sleep. For V30 there is no DCDC available so additional don't apply here.

Reply:
sounds like these should be regular dt props, @aasinclair please review as well

Reply:
Agreed, should be handled as device specific properties, e.g:
https://github.com/zephyrproject-rtos/zephyr/blob/main/dts/bindings/regulator/raspberrypi%2Ccore-supply-regulator.yaml#L21-L24

Reply:
@aasinclair @gmarull modes are now removed from driver, few properties added instead